### PR TITLE
fix Issue 20567 returning the result of a constructor should be NRVO

### PR DESCRIPTION
**Pull request by WalterBright**
More problems than I expected with https://github.com/dlang/dmd/pull/17050#issuecomment-2543521807

### CODE REVIEW

File: compiler/src/dmd/declaration.d [hunk 1]
```diff
@@ -95,6 +95,7 @@ extern (C++) abstract class Declaration : Dsymbol
       enum ignoreRead = 2; // ignore any reads of AliasDeclaration
       enum nounderscore = 4; // don't prepend _ to mangled name
       enum hidden       = 8; // don't print this in .di files
+      enum nrvo = 0x10;      /// forward to fd.nrvo_var when generating code
```
Comment (thewilsonator) [inline compiler/src/dmd/declaration.d:98]: header update?

Comment (WalterBright) [reply compiler/src/dmd/declaration.d:98]: those enums are only in frontend.h

**Comment by TurkeyMan** [comment]
This doesn't address the bug I mentioned... this patch appears to deal with copy construction.
You should add the code in my issue as a unittest.

**Comment by WalterBright** [comment]
@TurkeyMan I found this bug while investigating the bug you mentioned. It's a part of fixing your issue, and should be done before addressing the rest of it. The test case for this problem is in this PR.

**Comment by WalterBright** [comment]
This fix is gating the fix to https://github.com/dlang/dmd/pull/17050#issuecomment-2543521807

**Review by thewilsonator — APPROVED** [review]

**Comment by thewilsonator** [comment]
>  It's a part of fixing your issue, and should be done before addressing the rest of it. The test case for this problem is in this PR.

In which case I think the commit title (PR title? not sure. Both is a safe bet) should not include the work `fix`.

**Comment by WalterBright** [comment]
This fixed issue 20567: https://github.com/dlang/dmd/issues/20567

**Comment by WalterBright** [comment]
@thewilsonator I had 20567 and 20576 reversed. Oops.

**Comment by thewilsonator** [comment]
That is definitely fair.

**Comment by WalterBright** [comment]
Also, the auto-posting to bugzilla is now broken, as it's now github issues.

**Pull request merged**